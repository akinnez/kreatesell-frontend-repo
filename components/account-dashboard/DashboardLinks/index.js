import Link from "next/link";
import { useRouter } from "next/router";
import styles from "./index.module.scss";
// links
const links = [
  { label: "Both", value: "both", url: "/account/dashboard" },
  { label: "Kreator", value: "kreator", url: "/account/dashboard/kreator" },
  {
    label: "Affiliate",
    value: "affiliate",
    url: "/account/dashboard/affiliate",
  },
];

const DashboardLinks = () => {
  const router = useRouter();

  return (
    <div className={styles.links}>
      {links.map((link) => (
        <Link key={link.label} href={link.url}>
          <a
            className={`${styles.link} ${
              router.pathname.endsWith(link.value) && styles.active
            }`}
          >
            {link.label}
          </a>
        </Link>
      ))}
    </div>
  );
};

export default DashboardLinks;
