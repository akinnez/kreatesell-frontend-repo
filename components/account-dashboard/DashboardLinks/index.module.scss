@use "public/css/colors";
@use "public/css/mixins";

.links {
  @include mixins.flex(row, flex-start, center);
  column-gap: 10px;
  margin-bottom: 30px;
}

.link {
  padding: 9px 15px;
  border: 1px solid #d9d9d9;
  border-radius: 8px;
  color: colors.$base-gray;
  font-weight: 400;
  font-size: 14px;

  &:hover {
    border-color: colors.$base-blue;
    color: colors.$base-blue;
  }
}

.active {
  border-color: colors.$base-blue;
  color: colors.$base-white;
  background-color: colors.$base-blue;

  &:hover {
    color: colors.$base-white;
  }
}
