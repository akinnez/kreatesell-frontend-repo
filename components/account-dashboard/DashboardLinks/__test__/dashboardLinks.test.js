import DashboardLinks from "..";
import renderer from "react-test-renderer";

it("renders a snapshot of <DashboardLinks />", () => {
  const board = renderer.create(<DashboardLinks />).toJSON();
  expect(board).toMatchSnapshot();
});
