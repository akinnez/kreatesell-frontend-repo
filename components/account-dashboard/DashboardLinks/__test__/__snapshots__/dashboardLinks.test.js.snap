// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of <DashboardLinks />  1`] = `
<div
  className="links"
>
  <a
    className="link false"
    href="/account/dashboard"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    All
  </a>
  <a
    className="link active"
    href="/account/dashboard/kreator"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Kreator
  </a>
  <a
    className="link false"
    href="/account/dashboard/affiliate"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Affiliate
  </a>
</div>
`;
